packages/react-ai/src/components/AIConversation/views/Controls/MessagesControl.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import {
+  MessagesContext,
+  MessageVariantContext,
+  RoleContext,
+} from '../../context';
+import { AIConversationElements } from '../../context/elements';
+import { convertBufferToBase64, formatDate } from '../../utils';
+import { ActionsBarControl } from './ActionsBarControl';
+import { AvatarControl } from './AvatarControl';
+import { ConversationMessage } from '../../../../types';
+import { ResponseComponentsContext } from '../../context/ResponseComponentsContext';
+import { ControlsContext } from '../../context/ControlsContext';
+
+const { Image, Span, Text, View } = AIConversationElements;
+
+const MESSAGES_BLOCK = 'ai-messages';
+const MESSAGE_BLOCK = 'ai-message';
+
+const MediaContentBase = withBaseElementProps(Image, {
+  alt: 'Image attachment',
+});
+
+const MediaContent: typeof MediaContentBase = React.forwardRef(
+  function MediaContent(props, ref) {
+    const variant = React.useContext(MessageVariantContext);
+    const role = React.useContext(RoleContext);
+    return (
+      <MediaContentBase
+        {...props}
+        ref={ref}
+        className={`${MESSAGE_BLOCK}__image ${MESSAGE_BLOCK}__image--${variant} ${MESSAGE_BLOCK}__image--${role}`}
+      />
+    );
+  }
+);
+
+const TextContent: typeof Text = React.forwardRef(
+  function TextContent(props, ref) {
+    return <Text {...props} ref={ref} className={`${MESSAGE_BLOCK}__text`} />;
+  }
+);
+
+const ContentContainer: typeof View = React.forwardRef(
+  function ContentContainer(props, ref) {
+    const variant = React.useContext(MessageVariantContext);
+    return (
+      <View
+        data-testid={'content'}
+        {...props}
+        ref={ref}
+        className={`${MESSAGE_BLOCK}__content ${MESSAGE_BLOCK}__content--${variant}`}
```

Comment:
Haven't been looking closely at the ui components but we need to ensure that if the classname is provided in the props it overrides our classname

Reply:
Good catch, fixed this one, will audit the others